crates/interfaces/src/test_utils/full_block.rs

```diff
@@ -87,80 +112,120 @@ impl TestFullBlockClient {
 
     /// Get the block with the highest block number.
     pub fn highest_block(&self) -> Option<SealedBlock> {
-        let headers = self.headers.lock();
-        let (hash, header) = headers.iter().max_by_key(|(hash, header)| header.number)?;
-        let bodies = self.bodies.lock();
-        let body = bodies.get(hash)?;
-        Some(SealedBlock::new(header.clone().seal(*hash), body.clone()))
+        self.headers.lock().iter().max_by_key(|(_, header)| header.number).and_then(
+            |(hash, header)| {
+                self.bodies
+                    .lock()
+                    .get(hash)
+                    .map(|body| SealedBlock::new(header.clone().seal(*hash), body.clone()))
+            },
+        )
     }
 }
 
 impl DownloadClient for TestFullBlockClient {
+    /// Reports a bad message from a specific peer.
     fn report_bad_message(&self, _peer_id: PeerId) {}
 
+    /// Retrieves the number of connected peers.
+    ///
+    /// Returns the number of connected peers in the test scenario (1).
     fn num_connected_peers(&self) -> usize {
         1
     }
 }
 
+/// Implements the `HeadersClient` trait for the `TestFullBlockClient` struct.
 impl HeadersClient for TestFullBlockClient {
+    /// Specifies the associated output type.
     type Output = futures::future::Ready<PeerRequestResult<Vec<Header>>>;
 
+    /// Retrieves headers with a given priority level.
+    ///
+    /// # Arguments
+    ///
+    /// * `request` - A `HeadersRequest` indicating the headers to retrieve.
+    /// * `_priority` - A `Priority` level for the request.
+    ///
+    /// # Returns
+    ///
+    /// A `Ready` future containing a `PeerRequestResult` with a vector of retrieved headers.
     fn get_headers_with_priority(
         &self,
         request: HeadersRequest,
         _priority: Priority,
     ) -> Self::Output {
+        // Locks the headers for concurrent access.
```

Comment:
I don't think this additional comment is very useful.

Reply:
removed